http-server/src/main/java/io/airlift/http/server/HttpServerModule.java

```diff
@@ -57,32 +61,56 @@
 public class HttpServerModule
         extends AbstractConfigurationAwareModule
 {
+    private final Optional<Class<? extends Annotation>> qualifier;
+    private final String configPrefix;
+
+    public HttpServerModule(@Nullable Class<? extends Annotation> qualifier, @Nullable String configPrefix)
+    {
+        this.qualifier = Optional.ofNullable(qualifier);
+        this.configPrefix = configPrefix;
+    }
```

Comment:
When the qualifier is set, I assume the config prefix is required?  If so maybe we should have a check for that?

@electrum should we use a named static factory to make the intention more clear here?  I could go either way.